commands.c

```diff
@@ -81,14 +82,69 @@ command_answer *send_command(char *name, char *data, task_context *context)
     return cmd_answer;
 }
 
+command_answer *send_attest_command(direction direction, struct sock *s, u16 port)
+{
+    int iplen = 16;
+    char ipformat[] = "%pI4";
+
+    if (s->sk_family == AF_INET6)
+    {
+        iplen = 46;
+        strcpy(ipformat, "%pI6");
+    }
+
+    unsigned char source_ip[iplen];
+    u16 source_port;
+    unsigned char destination_ip[iplen];
+    u16 destination_port;
+
+    if (direction == INPUT)
+    {
+        snprintf(source_ip, sizeof(source_ip), ipformat, &s->sk_daddr);
+        snprintf(destination_ip, sizeof(destination_ip), ipformat, &s->sk_rcv_saddr);
+        source_port = s->sk_dport;
+        destination_port = s->sk_num;
+    }
+    else
+    {
+        snprintf(source_ip, sizeof(source_ip), ipformat, &s->sk_rcv_saddr);
+        snprintf(destination_ip, sizeof(destination_ip), ipformat, &s->sk_daddr);
+        source_port = s->sk_num;
+        destination_port = port;
+    }
+
+    JSON_Value *root_value = json_value_init_object();
+    JSON_Object *root_object = json_value_get_object(root_value);
+
+    json_object_set_number(root_object, "direction", direction);
+    json_object_set_string(root_object, "source_ip", source_ip);
+    json_object_set_number(root_object, "source_port", source_port);
+    json_object_set_string(root_object, "destination_ip", destination_ip);
+    json_object_set_number(root_object, "destination_port", destination_port);
+
+    char *serialized_string = json_serialize_to_string(root_value);
+
+    command_answer *answer = send_command("attest", serialized_string, get_task_context());
+
+    json_free_serialized_string(serialized_string);
+    json_value_free(root_value);
```

Comment:
This json_value_free function name is misleading for me. I would add a command that this function also frees the root_object field.